types/jest/index.d.ts

```diff
@@ -72,6 +72,8 @@ type ExtractEachCallbackArgs<T extends ReadonlyArray<any>> = {
         : 'fallback'
 ];
 
+type MockModuleFactory<T> = () => T extends { default: any } ? Partial<T> & { __esModule: boolean } : Partial<T>;
```

Comment:
I don’t think that `__esModule` should be mandatory.
```suggestion
type MockModuleFactory<T> = () => Partial<T>;
```

Reply:
The idea is was that if a module had a `default` key, then and only then `__esModule` is mandatory (could still provide false).
As we don't know if the Dev is overriding the `default` or not, thought it would be good to encourage them to set the value explicitly. The value is required by Jest when the `default` is being replaced.

Open to other suggestions or could remove if people disagree.